### ISSUE TITLE
USTC-Pickers in SeisBench

### ISSUE THREAD

**Issue by yetinam** [JUNZHU-SEIS/USTC-Pickers#1]
Hi @JUNZHU-SEIS ,

I've read your article on the USTC-Pickers. Great work! I was wondering if you'd be interested in making the model weights available through SeisBench? I'd probably only put the model trained on all of DiTing in and refer to your repository for the individual pickers. Clearly, I'd reference your publication in the weights docstring.

Let me know if you're interested in this. If yes, I'd be happy to integrate the weight and make them easily obtainable for all SeisBench users.

**Comment by JUNZHU-SEIS** [comment]
Hi @yetinam ,
That is great. [Here](https://github.com/JUNZHU-SEIS/USTC-Pickers/blob/phasenet_fix/model_list/%E4%B8%AD%E5%9B%BD.pt) is the model file trained on all of DiTing. It has the same architecture as PhaseNet (NOT PhaseNetLight).  Please let me know if you need any help.

**Comment by yetinam** [comment]
Great! Just two more questions:
- How did you normalize the inputs when training? Using the peak or the std of the waveform? And did you normalize each individual component or all components jointly?
- Did you use any frequency-based filtering on the waveforms when training?

**Comment by JUNZHU-SEIS** [comment]
* I normalize individual components by the std
* No filtering for the input waveforms. But I use a bandpass filter of 1-15 Hz when augmentation applied (e.g., filtering the input waveforms if it is clipped, filtering the superimposed event signal before stacking).

Below is the normalization function:

```python
def normalize(window):
	window = window - np.mean(window, axis=-1, keepdims=True)
	std = np.std(window, axis=-1, keepdims=True)
	std[std == 0] = 1  # Avoid NaN errors
	window = window / std
	return window
```

**Comment by yetinam** [comment]
Thanks for the quick reply. The picker is now available through SeisBench. Just use `sbm.PhaseNet.from_pretrained("diting")` to give it a try.

As a sidenote, when using the weights I noticed that the tensors are by default mapped to a CUDA device. This means they can not be loaded without further tricks on a machine without CUDA. If you want to change this and make the weights directly loadable from SeisBench, the following code does the trick:
```python
model = torch.load(GPU_PATH, map_location=torch.device('cpu'))
torch.save(model.state_dict(), CPU_PATH)
```

And if you want to add the json-files required to use `sbm.PhaseNet.load`, you can just adapt [this file](https://dcache-demo.desy.de:2443/Helmholtz/HelmholtzAI/SeisBench/models/v3/phasenet/diting.json.v1).

Thanks again for contributing the model to SeisBench!

**Comment by yetinam** [comment]
Hi, I got a quick question regarding the weights discussed in this issue. On what sampling rate did you train the model? I assumed 100 Hz, but it was lately [pointed out](https://github.com/seisbench/seisbench/issues/307) to me that it's probably 50 Hz.

**Comment by JUNZHU-SEIS** [comment]
Yes, it is 50 Hz.

**Comment by yetinam** [comment]
Thanks, I fixed it now.
